### ISSUE TITLE
running Gunicorn with Django Channels2 installed in library got RuntimeError  

### ISSUE THREAD

**Issue by DrChai** [getsentry/sentry-python#515]
I have one Django project that is running behind Gunicorn(wsgi) and Channels2(asgi) for handling HTTP and WebSocket separately. 
Running Gunicorn would raise a [RuntimeError](https://github.com/getsentry/sentry-python/blob/534757568767dca79f623309cf5c529fc505e81e/sentry_sdk/integrations/django/__init__.py#L285https://github.com/getsentry/sentry-python/blob/534757568767dca79f623309cf5c529fc505e81e/sentry_sdk/integrations/django/__init__.py#L285): 
` RuntimeError(
            "We detected that you are using Django channels 2.0. To get proper "
            "instrumentation for ASGI requests, the Sentry SDK requires "
            "Python 3.7+ or the aiocontextvars package from PyPI."
        )`
Either upgrading to py3.7 or Installing `aiocontextvars` would not solve the problem.
It looks that `_patch_channels()` https://github.com/getsentry/sentry-python/blob/534757568767dca79f623309cf5c529fc505e81e/sentry_sdk/integrations/django/__init__.py#L276 will be called anyway during initialization, and `HAS_REAL_CONTEXTVARS` is False because of gevent is installed
Related libraries and versions:
[packages]

Django==2.2.3
channels==2.3.0
aiocontextvars==0.2.2
gevent==1.4.0
gunicorn==19.9.0

[requires]
python_version = "3.6"

**Comment by untitaker** [comment]
I believe you are calling `gevent.monkey.patch_all()` even for your websockets server, is that correct?

**Comment by DrChai** [comment]
@untitaker No, i just run Gunicorn with gevent worker class regularly:
` gunicorn -c gunicorn_config.py myapp.wsgi`
and gunicorn_config.py:
`workers = 2
timeout = 30
worker_class = 'gevent'
max_requests = 1000
loglevel = 'info'
bind = "0.0.0.0:8000"`

**Comment by untitaker** [comment]
Yes, for your regular HTTP requests, but what thing serves your websocket requests?

**Comment by untitaker** [comment]
Concretely, are you using Daphne or some other ASGI server to serve websockets?

**Comment by DrChai** [comment]
@untitaker yes, using Gunicorn for Http request and using Daphne(Channels2) for Websocket requests. Those two services run from the same Docker image, that is, two commands with the same env, libraries, and Django code. No error detected from the daphne service.

I guess i have to use two different Dockerfile for Daphne and Gunicorn, explicitly exclude channels library from my Gunicorn's Dockerfile.

**Comment by untitaker** [comment]
That's not necessary, although it probably would solve the problem! Do you
import gevent anywhere in your code explicitly?

**Comment by DrChai** [comment]
@untitaker, No never import gevent in my code

**Comment by untitaker** [comment]
Can you tell me the value of _is_threading_local_monkey_patched
In the process where channels runs? For some reason the sdk believes
contextvars are not safe to use, and it seems either eventlet or gevent
claims to have patched threading. So something must have called
gevent.monkey.patch_all (since eventlet is entirely absent)

**Comment by DrChai** [comment]
@untitaker By interrupting the process of the service, _is_threading_local_monkey_patched returns True.
now i can successfully reproduce this error on an empty Django project created by django-admin. Here is relative packages installed:
```
[packages]
django = "*"
gunicorn = {extras = ["gevent"],version = "*"}
sentry-sdk = "*"
channels = "*"
aiocontextvars = "*"

[requires]
python_version = "3.6"
```
just running Gunicorn with the command: 
`gunicorn -b 0.0.0.0:8000 -w 1 -k gevent porject.wsgi`

**Comment by untitaker** [comment]
Oh, sorry! I understand now. I thought it was Daphne that was crashing. But it's gunicorn, which makes sense. Will fix this week.

In the meantime please downgrade the SDK to 0.11

**Comment by untitaker** [comment]
This will be fixed in 0.12.3
